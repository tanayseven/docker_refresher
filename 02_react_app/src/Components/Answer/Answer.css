.Answer button {
    font-size: 25px;
    display: flex;
    width: 100%;
    letter-spacing: 2px;
    margin-bottom: 10px;
    padding: 8px 10px;
    color: #fff;
    border: 1px transparent solid;
}

.Answer button:hover {
    -webkit-tap-highlight-color: transparent;
    color: #dce869;
    border-color: #dce869;
}

.Answer button:active {
    -webkit-tap-highlight-color: transparent;
    color: #dce869;
    border-color: #dce869;
}

.Answer button:focus{
    -webkit-tap-highlight-color: transparent;
    color: #dce869;
    border-color: #dce869;
}

.Answer button span {
    padding-right: 10px;
}