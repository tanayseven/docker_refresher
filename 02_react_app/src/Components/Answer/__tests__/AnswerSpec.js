import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Answer from './../index';

describe('Answer', () => {
    let answerProps;
    beforeEach(() => {
        answerProps = {
            value: 12,
            currentIndex: 1,
            handleClick: jest.fn()
        };
    })
    it('renders correctly', () => {
        let tree = renderer.create(
            <Answer {...answerProps} />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('click should call passedin click handler', () => {
        let mockClickHandler = jest.fn();
        answerProps.handleClick = mockClickHandler;
        let wrapper = shallow(<Answer {...answerProps} />);
        let elm = wrapper.find('.Answer--Btn')
        elm.simulate("click");
        expect(mockClickHandler).toHaveBeenCalled()
    })
});