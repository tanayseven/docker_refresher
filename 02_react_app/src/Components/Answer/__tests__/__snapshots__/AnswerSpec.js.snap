// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer renders correctly 1`] = `
<div
  className="Answer"
>
  <button
    className="Answer--Btn"
    onClick={[Function]}
    onTouchStart={[Function]}
  >
    <span>
      1
      ) 
    </span>
     
    12
  </button>
</div>
`;
